test/unit/test_array.cxx

```diff
@@ -436,6 +436,29 @@ void test_array_generate_empty_strings()
     "Array of 12 empty strings came out wrong.");
 }
 
+void test_sparse_arrays()
+{
+  // Reproduce #922 : NULL not paying for its separator in an array, causing
+  // problems in sparse arrays
+
+  PQXX_CHECK_EQUAL(
+    pqxx::to_string(std::vector<std::optional<int>>{
+      std::nullopt, std::nullopt, std::nullopt, std::nullopt}),
+    "{NULL,NULL,NULL,NULL}",
+    "Array of optional<int> filled with std::nullopt came out wrong");
+
+  std::array<std::optional<int>, 14> sparseArray{std::nullopt, std::nullopt,
```

Comment:
I think you can just write `std::array<std::optional<int>, 14>` and get null values by default!